lib/Conversion/TorchToLinalg/Uncategorized.cpp

```diff
@@ -2336,6 +2336,305 @@ class ConvertCastEquivalentOp : public OpConversionPattern<OpTy> {
+    Value notValidEmpty_0 =
+        rewriter.create<tensor::EmptyOp>(loc, gridShapeExtracted, int1type);
+    Value notValidEmpty_1 =
+        rewriter.create<tensor::EmptyOp>(loc, gridShapeExtracted, int1type);
+
+    SmallVector<AffineMap> gridMaps(10,
+                                    rewriter.getMultiDimIdentityMap(gridRank));
+    SmallVector<utils::IteratorType> gridIterators(
+        gridRank, utils::IteratorType::parallel);
+    auto gridType_0 = gridResultEmpty_0.getType().cast<RankedTensorType>();
+    auto lowerType_0 = lowerEmpty_0.getType().cast<RankedTensorType>();
+    auto validType_0 = notValidEmpty_0.getType().cast<RankedTensorType>();
+
+    auto sGrid = rewriter.create<linalg::GenericOp>(
+        loc,
+        TypeRange{gridType_0, gridType_0, lowerType_0, lowerType_0, lowerType_0,
+                  lowerType_0, validType_0, validType_0},
+        ValueRange{grid_0, grid_1},
+        ValueRange{gridResultEmpty_0, gridResultEmpty_1, lowerEmpty_0,
+                   lowerEmpty_1, upperEmpty_0, upperEmpty_1, notValidEmpty_0,
+                   notValidEmpty_1},
+        gridMaps, gridIterators,
+        [&](OpBuilder &b, Location loc, ValueRange args) {
+          Value gr_0 = args[0];
+          Value gr_1 = args[1];
+          Value oneFloat = b.create<arith::ConstantOp>(
+              loc, b.getFloatAttr(gridElementType, 1.0));
+          Value gplus_0 = b.create<arith::AddFOp>(loc, gr_0, oneFloat);
+          Value gplus_1 = b.create<arith::AddFOp>(loc, gr_1, oneFloat);
+          Value result_0 = b.create<arith::MulFOp>(loc, gplus_0, innerDim_0e);
+          Value result_1 = b.create<arith::MulFOp>(loc, gplus_1, innerDim_1e);
+          Value lower_0 = b.create<arith::FPToSIOp>(loc, int64type, result_0);
+          Value lower_1 = b.create<arith::FPToSIOp>(loc, int64type, result_1);
+          Value oneInt = rewriter.create<arith::ConstantOp>(
+              loc, rewriter.getIntegerAttr(int64type, 1));
+          Value upper_0 =
+              b.create<arith::AddIOp>(loc, int64type, lower_0, oneInt);
+          Value upper_1 =
+              b.create<arith::AddIOp>(loc, int64type, lower_1, oneInt);
+          Value notValid_0 =
+              createGreaterThan(rewriter, loc, int64type, upper_0, innerDim_0c);
+          Value notValid_1 =
+              createGreaterThan(rewriter, loc, int64type, upper_1, innerDim_1c);
+          Value upperValid_0 =
+              b.create<arith::SelectOp>(loc, notValid_0, lower_0, upper_0);
+          Value upperValid_1 =
+              b.create<arith::SelectOp>(loc, notValid_1, lower_1, upper_1);
+          b.create<linalg::YieldOp>(loc, ValueRange{result_0, result_1, lower_0,
+                                                    lower_1, upperValid_0,
+                                                    upperValid_1, notValid_0,
+                                                    notValid_1});
+        });
+
+    Value gridResult_0 = rewriter.create<tensor::CastOp>(
```

Comment:
You should not require casts here. If you are something is going wrong with how you are selecting the expected types. Everything between linalg ops should be shape compatible.